### PR TITLE
MTDSA-12892: Update common code for TY 23-24 API#1860(1670)Employment Expenses API 

### PR DESCRIPTION
**Pull request by darshanbala-hmrc**

### CODE REVIEW

File: test/v1/connectors/BaseDownstreamConnectorSpec.scala [hunk 1]
```diff
@@ -50,6 +52,7 @@ class BaseDownstreamConnectorSpec extends ConnectorSpec {
     MockAppConfig.desToken returns "des-token"
     MockAppConfig.desEnvironment returns "des-environment"
     MockAppConfig.desEnvironmentHeaders returns desEnvironmentHeaders
+    MockAppConfig.featureSwitches returns Configuration("tys-api.enabled" -> false)
```
Comment (sean-flynn-HMRC) [inline test/v1/connectors/BaseDownstreamConnectorSpec.scala:55]: Is it necessary to set tys-enabled to false? The tests pass without doing this. Also, the template in individuals-income-received api doesn't do this step either

Comment (darshanbala-hmrc) [reply test/v1/connectors/BaseDownstreamConnectorSpec.scala]: Verified tests do in fact pass without them. Removed and updated. Thanks!

File: app/v1/models/errors/desErrors.scala [hunk 2]
```diff
@@ -26,12 +26,12 @@ object DesErrorCode {
   implicit val reads: Reads[DesErrorCode] = Json.reads[DesErrorCode]
```
Comment (sean-flynn-HMRC) [inline app/v1/models/errors/desErrors.scala:26]: Curious as to why we've kept the "DesErrorCode" stuff on lines 21-26?

File: app/v1/support/DownstreamResponseMappingSupport.scala [hunk 3]
```diff
@@ -16,34 +16,44 @@
...
+trait DownstreamResponseMappingSupport {
   self: Logging =>
 
-  final def mapDesErrors[D](errorCodeMap: PartialFunction[String, MtdError])(desResponseWrapper: ResponseWrapper[DesError])(implicit
-      logContext: EndpointLogContext): ErrorWrapper = {
+  final def validateRetrieveResponse[T: Writes](downstreamResponseWrapper: ResponseWrapper[T]): Either[ErrorWrapper, ResponseWrapper[T]] = {
+    if (Json.toJson(downstreamResponseWrapper.responseData) == JsObject.empty) {
+      Left(ErrorWrapper(downstreamResponseWrapper.correlationId, NotFoundError, None))
+    } else {
+      Right(downstreamResponseWrapper)
```
Comment (sean-flynn-HMRC) [inline app/v1/support/DownstreamResponseMappingSupport.scala:33]: I think you've improved on the template here. Nice to see no desResponseWrappers!

File: app/v1/models/errors/desErrors.scala (hunk 2 above)
Comment (darshanbala-hmrc) [reply app/v1/models/errors/desErrors.scala]: Fixed

**Review by sean-flynn-HMRC — APPROVED** [review]
LGTM

**Pull request merged**